tests/test_main.py

```diff
@@ -25,23 +31,33 @@ def test_run(mongo_database, testing_files_for_process, pyodbc_conn):
     with patch("crawler.file_processing.CentreFile.insert_samples_from_docs_into_mlwh"):
         run(False, False, False, "crawler.config.integration")
 
-    # We expect to have three collections following import
+    # We expect to have four collections following import
     centres_collection = get_mongo_collection(mongo_database, COLLECTION_CENTRES)
     imports_collection = get_mongo_collection(mongo_database, COLLECTION_IMPORTS)
     samples_collection = get_mongo_collection(mongo_database, COLLECTION_SAMPLES)
+    source_plates_collection = get_mongo_collection(mongo_database, COLLECTION_SOURCE_PLATES)
 
     # We record our test centres
     assert centres_collection.count_documents({}) == NUMBER_CENTRES
     assert centres_collection.count_documents({"name": "Test Centre"}) == 1
 
-    # We record *all* our samples
+    # We record all our source plates
```

Comment:
Maybe here could double check the uuids have been created here too

Reply:
If we do that, I think we further loose the separation that unit tests should give us. I already don't really like that we test the functionality of other classes/methods (adding the source plates/samples) in most test, making them hybrid integration tests. Unit tests should really only test what happens in the method being tested.

I already test the uuid creation logic in the file_processing tests, and I think a fair compromise is having testing that more complex logic stay in just the one place